.github/workflows/ci.yml

```diff
@@ -31,6 +31,18 @@ jobs:
       - uses: swatinem/rust-cache@v1
       - run: cargo check
 
+  check-lock-diff-check:
+    name: check-lock-diff-check
+    runs-on: ubuntu-latest
+    needs: check
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@1.60.0
+      - uses: swatinem/rust-cache@v1
+      - run: cargo generate-lockfile
+      - run: git status --porcelain | grep -q Cargo.lock && echo "Lockfile has diff" && exit 1
```

Comment:
A better solution for this might be: `cargo metadata --format-version 1 >/dev/null` which seems to exit with nonzero status if the lockfile is not up to date (I have to confirm this).

Reply:
There is the `--locked` which could be used with for example `cargo update`: `cargo update --locked`.

That fails for me if the Cargo.toml is out of date.